framework/assets/yii.validation.js

```diff
@@ -244,7 +244,15 @@ yii.validation = (function ($) {
 
             value = $input.val();
             if (!options.skipOnEmpty || !pub.isEmpty(value)) {
-                value = $.trim(value);
+                value = new String(value);
+                if (options.chars || !String.prototype.trim) {
+                    var chars = !options.chars
```

Comment:
Maybe extract this logic to a separate function without extra dependencies? Can be reused in `required`, etc.

Reply:
using where?